### PR TITLE
GN-168 Back to top link

### PR DESCRIPTION
**Pull request by johndavey72**
https://nicedigital.atlassian.net/browse/GN-168

See also https://github.com/nice-digital/bnf-gatsby/pull/17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -637,6 +640,27 @@ The displayName is the user's name and must be provided if the "idam" provider i
 The identifier of the currently active service.
 See [services.json](src/services.json) for a list of the available service identifiers.
 
+###### Footer.backToTop
```
Comment (ediblecode) [inline README.md:643]: I don't think we have this option any more?

File: examples/examples.js [hunk 2]
```diff
@@ -118,6 +118,12 @@ const bnfConfig = {
 					window.location.hash = "Search?q=" + encodeURIComponent(searchArgs.query);
 				}
 			}
+		},
+		footer: {
+			backToTop: {
```
Comment (ediblecode) [inline examples/examples.js:123]: Same comment as above, I don't think we need these options here

File: src/Main/BackToTop/BackToTop.test.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		const wrapper = shallow(<BackToTop service="bnf" />);
+		expect(wrapper.find("a[href='#top']").length).toEqual(1);
+	});
+
+	it("Renders a link with the correct hash when a scroll target id prop provided and target exists", () => {
+		let scrollTargetDiv = document.createElement("div");
+		scrollTargetDiv.id = "content-start";
+		document.body.appendChild(scrollTargetDiv);
+		const wrapper = shallow(
+			<BackToTop service="bnf" scrollTargetId="content-start" />
```
Comment (ediblecode) [inline src/Main/BackToTop/BackToTop.test.jsx:28]: These props and this test don't make sense any more

File: src/Main/BackToTop/BackToTop.test.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import toJson from "enzyme-to-json";
+
+describe("BackToTop", () => {
+	it("Renders without crashing", () => {
+		const wrapper = shallow(<BackToTop />);
+		expect(wrapper).toHaveLength(1);
+	});
+
+	it("Matches snapshot", () => {
+		const wrapper = shallow(<BackToTop service="bnf" />);
```
Comment (ediblecode) [inline src/Main/BackToTop/BackToTop.test.jsx:14]: Can you remove the service prop from each of these?

File: src/Main/Main.module.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+@supports (position: sticky) {
+  .main {
+    padding-bottom: 4rem;
```
Comment (ediblecode) [inline src/Main/Main.module.scss:3]: Can this have a comment to explain it's a space for the back to top link to dock? Also, on CKS we use the `$back-to-top-height` variable. I'm not against just using 4rem but maybe a NDS spacing var would be better, and with a comment. Either way, without that you just have a magic number of 4rem which has significance

File: src/Header/Header.jsx [hunk 6]
```diff
@@ -95,7 +95,7 @@ export class Header extends Component {
 						}}
 					</HeaderContext.Consumer>
 
-					<div className={styles.header} data-tracking="Global nav">
+					<div className={styles.header} data-tracking="Global nav" id="top">
```
Comment (ediblecode) [inline src/Header/Header.jsx:98]: Should probably have a test that asserts that we have a back to top scroll target id

File: src/Footer/__snapshots__/Footer.test.jsx.snap [hunk 7]
```diff
@@ -1,31 +1,33 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Legal Matches snapshot 1`] = `
-<footer
-  className="footer"
-  data-tracking="Global footer"
->
-  <div
-    className="container"
+<Fragment>
```
Comment (ediblecode) [inline src/Footer/__snapshots__/Footer.test.jsx.snap:4]: Not sure this test will pass will it? We're not returning a fragment here

**Review by ediblecode — CHANGES_REQUESTED** [review]

File: src/Footer/__snapshots__/Footer.test.jsx.snap (hunk 7 above)
Comment (johndavey72) [reply src/Footer/__snapshots__/Footer.test.jsx.snap:4]: updated the snapshot, seems to have sorted it.

**Review by ediblecode — APPROVED** [review]

**Pull request merged**